### ISSUE TITLE
RequestPolicy should completely take subscription data from the subscription files (json files)

### ISSUE THREAD

**Issue by myrdd** [RequestPolicyContinued/requestpolicy#678]
Currently the available subscriptions are hard-coded, and also their titles and descriptions. Instead, the only thing hard-coded should be the URL to [`official.json`](https://github.com/RequestPolicyContinued/subscriptions/blob/master/official.json).

What should be extracted from `official.json`?

 * the actual list of subscriptions with their URL
 * their title
 * their description

In addition, if the URL is the only available information for one subscription, it should still be visible on `about:requestpolicy?subscriptions`.

**Comment by acrobat1** [comment]
This is a dumb question (I know) but by *hard-coded* do we actually mean that in the current state of RP official subscriptions are never going to be automatically updated?

If that's the case, can you simply copy the up-to date list(s) from https://github.com/RequestPolicyContinued/subscriptions/  to the requestpolicy directory inside your Firefox profile?

**Comment by nodiscc** [comment]
@acrobat1 the subscriptions are updated automatically.

What is hardcoded -> RP would only use, update and download `official-allow_functionality.json official-allow_sameorg.json official-deny_trackers.json` even if we decided to add another extra subscription to [official.json](https://github.com/RequestPolicyContinued/subscriptions/blob/master/official.json)

**Comment by acrobat1** [comment]
I should probably open a new issue then, in short the <code> deny_trackers.json </code>list I have in my profile is much shorter then the current list  available on  https://github.com/RequestPolicyContinued/subscriptions/commit/994ba7dd3835f3123416eef26eafc6b2fb04fb13   and its modification date is 2014-05-16, apparently it has never been updated since I've first installed RP.

That is, unless I'm not understanding at all what  we are talking about, and/or I've inadvertently blocked the IP address from which such lists should be automatically updated.

**Comment by myrdd** [comment]
> I should probably open a new issue then, in short the deny_trackers.json list I have in my profile is much shorter then the current list available on RequestPolicyContinued/subscriptions@994ba7d and its modification date is 2014-05-16, apparently it has never been updated since I've first installed RP.

@acrobat1 the issue here could be that you've configured RP for "default deny" mode. In that mode the `deny_trackers.json` is disabled (and it's not possible to enable it, see #529). Since it's not enabled, it doesn't get updated.

**Comment by acrobat1** [comment]
Yes, you are right: I wasn't aware of that, but now thinking of it somehow it does make sense .
If *everyting* is blocked by default, there is no need to block trackers -still, why not update the  lists anyways, in case one toggles that preference off again?

**Comment by myrdd** [comment]
> still, why not update the lists anyways, in case one toggles that preference off again?

When the "default policy" is toggled, the subscription will be updated – at least that's what I suppose. In any case, #529 will allow to enable any subscription at any time. With that additional subscription, it could be possible to show in the menu which destinations might be trackers according to the "trackers" subscription.
